src/test/java/htsjdk/samtools/cram/CRAMCodecCorpusTest.java

```diff
@@ -0,0 +1,343 @@
+                Assert.assertEquals(ransDecode.uncompress(compressedBytes), uncompressedBytes);
+            }
+        }
+    }
+
+    @Test (
+            dataProvider = "allRansCodecsAndData",
+            dependsOnMethods = "testGetHTSCodecsCorpus",
+            description = "Compress the original file using htsjdk RANS and compare it with the existing compressed file. " +
+                    "Uncompress the existing compressed file using htsjdk RANS and compare it with the original file.")
+    public void testRANSPreCompressed(
+            final Path inputTestDataPath,
+            final RANSEncode unused,
+            final RANSDecode ransDecode,
+            final RANSParams params,
+            final String CompressedDirname) throws IOException {
+        if (!CRAMCodecCorpus.isHtsCodecsTestDataAvailable()) {
+            throw new SkipException("htscodecs test data is not available locally");
+        }
+
+        final Path preCompressedDataPath = getCompressedRANSPath(CompressedDirname,inputTestDataPath, params);
+
+        try (final InputStream inputStream = Files.newInputStream(inputTestDataPath);
+             final InputStream preCompressedInputStream = Files.newInputStream(preCompressedDataPath);
+        ) {
+            // preprocess the uncompressed data (to match what the htscodecs-library test harness does)
+            // by filtering out the embedded newlines, and then round trip through RANS and compare the
+            // results
+            final ByteBuffer inputBytes = ByteBuffer.wrap(filterEmbeddedNewlines(IOUtils.toByteArray(inputStream)));
+
+            final ByteBuffer preCompressedInputBytes = ByteBuffer.wrap(IOUtils.toByteArray(preCompressedInputStream));
+
+            // commenting as htsjdkCompressedBytes is not used anywhere
+            // Use htsjdk to compress the input file from htscodecs repo
+//            final ByteBuffer htsjdkCompressedBytes = ransEncode.compress(inputBytes, params);
+//            inputBytes.rewind();
+
+//            // commenting as the comparison of compressed bytes is not needed to ensure interoperability.
+//            // Compare the htsjdk compressed bytes with the precompressed file from htscodecs repo
+//            Assert.assertEquals(htsjdkCompressedBytes, preCompressedInputBytes);
+
+            // Use htsjdk to uncompress the precompressed file from htscodecs repo
+            final ByteBuffer htsjdkUncompressedBytes = ransDecode.uncompress(preCompressedInputBytes);
+
+            // Compare the htsjdk uncompressed bytes with the original input file from htscodecs repo
+            Assert.assertEquals(htsjdkUncompressedBytes, inputBytes);
+        } catch (NoSuchFileException ex){
+            // if precompressed file or input file is not present
+            System.out.println("Skipping testRANSPrecompressed as either input file " +
+                    "or precompressed file is missing. File Missing: " + ex.getMessage());
```

Comment:
Got it! I will replace the `catch` block with `throws`. But, should the `try` block be removed as well since, it limits the scope of `inputStream` and `preCompressedInputStream` to this block of code?

Reply:
On second thought, adding `throw new SkipException()` inside the `catch` block as the test is expected to be skipped when the input file is missing.